src/pages/EnablePayments/IdologyQuestions.js

```diff
@@ -54,149 +51,121 @@ const defaultProps = {
     walletAdditionalDetails: {},
 };
 
-class IdologyQuestions extends React.Component {
-    constructor(props) {
-        super(props);
-        this.submitAnswers = this.submitAnswers.bind(this);
+function IdologyQuestions({questions, walletAdditionalDetails, idNumber}) {
+    const formRef = useRef();
+    const {translate} = useLocalize();
 
-        this.state = {
-            /** Current question index to display. */
-            questionNumber: 0,
+    const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+    const [shouldHideSkipAnswer, setShouldHideSkipAnswer] = useState(false);
+    const [userAnswers, setUserAnswers] = useState([]);
+    const [error, setError] = useState('');
 
-            /** Should we hide the "Skip question" answer? Yes if the user already skipped MAX_SKIP questions. */
-            hideSkip: false,
-
-            /** Answers from the user */
-            answers: [],
+    const currentQuestion = questions[currentQuestionIndex] || {};
+    const possibleAnswers = _.filter(
+        _.map(currentQuestion.answer, (answer) => {
+            if (shouldHideSkipAnswer && answer === SKIP_QUESTION_TEXT) {
+                return;
+            }
 
-            /** Any error message */
-            errorMessage: '',
-        };
-    }
+            return {
+                label: answer,
+                value: answer,
+            };
+        }),
+    );
+    const errorMessage = ErrorUtils.getLatestErrorMessage(walletAdditionalDetails) || error;
 
     /**
      * Put question answer in the state.
-     * @param {Number} questionIndex
      * @param {String} answer
      */
-    chooseAnswer(questionIndex, answer) {
-        this.setState((prevState) => {
-            const answers = prevState.answers;
-            const question = this.props.questions[questionIndex];
-            answers[questionIndex] = {question: question.type, answer};
-            return {
-                answers,
-                errorMessage: '',
-            };
-        });
-    }
+    const chooseAnswer = (answer) => {
+        const tempAnswers = _.map(userAnswers, _.clone);
 
+        tempAnswers[currentQuestionIndex] = {question: currentQuestion.type, answer};
+
+        setUserAnswers(tempAnswers);
+        setError('');
+    };
```

Comment:
A space before comment block would be good to have
```suggestion
    };
    
```

Reply:
Added